qiskit_nature/problems/second_quantization/lattice/lattice_model_problem.py

```diff
@@ -0,0 +1,116 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""The Lattice Model Problem class."""
+
+from typing import Union, Callable, List, Tuple
+
+import numpy as np
+
+from qiskit.algorithms import EigensolverResult, MinimumEigensolverResult
+from qiskit_nature import ListOrDictType, settings
+from qiskit_nature.converters.second_quantization import QubitConverter
+from qiskit_nature.operators.second_quantization import SecondQuantizedOp
+from qiskit_nature.results import EigenstateResult, LatticeModelResult
+from .models.lattice_model import LatticeModel
+from ..base_problem import BaseProblem
+
+
+class LatticeModelProblem(BaseProblem):
+    """Lattice Model Problem class to create second quantized operators from a lattice model."""
+
+    def __init__(self, lattice_model=LatticeModel) -> None:
+        """
+        Args:
+            lattice_model: A lattice model class to create second quantized operators.
+        """
+        super().__init__()
+        self._lattice_model = lattice_model
+        self._main_property_name = "LatticeEnergy"
```

Comment:
I talked this with @ikkoham. We can pass a `main property_name` to a constructor of `BaseProblem` to set it. e.g. `super().__init__("LatticeEnergy")`. The comment is legit since `BaseProblem` has `_main_property_name` variable with an empty string. 

What do you think? @mrossinek 
I can change all problem subclasses to set their name like `super().__init__("LatticeEnergy")` if you want. Then, I will open a new PR for that change.

Reply:
Sorry, I overlooked this comment before. Sounds like a reasonable change to me :+1: Could you do this in a separate PR?